onnxruntime/core/graph/contrib_ops/quantization_defs.cc

```diff
@@ -31,16 +25,40 @@ using ONNX_NAMESPACE::OPTIONAL_VALUE;
 using ONNX_NAMESPACE::DbgOperatorSetTracker;
 #endif
 
-void ValidateTypeAndShapeForScaleAndZP(ONNX_NAMESPACE::InferenceContext& ctx, int index, ::google::protobuf::int32 expectedType, bool isScalar, int expectedTensorSize) {
+static void CheckFitsInsideBoundingBox(const ONNX_NAMESPACE::TensorShapeProto& input_shape,
+                                       const google::protobuf::RepeatedField<int64_t>& bounding_box_dims,
+                                       const BFPType& bfp_type) {
+  bool all_dim_values = true;
+  int64_t numel = 1;
+
+  for (auto i = 0; i < bounding_box_dims.size(); i++) {
+    auto dim = input_shape.dim(static_cast<int>(bounding_box_dims[i]));
+    if (!dim.has_dim_value()) {
+      all_dim_values = false;
+      break;
+    }
+    numel *= dim.dim_value();
+  }
+
+  auto is_custom_bfp = bfp_type == BFPType::Custom_BFP_0 || bfp_type == BFPType::Custom_BFP_0;
+
+  // all standard BFP types use a bounding box size of 16
```

Comment:
standard bfp types don't necessarily have bounding box of 16 - i need to map the type to a bounding box size.

Reply:
done.